### PR TITLE
Turtle dumping

### PR DESCRIPTION
**Pull request by BeanGalSolina**
please dump turtle programs into this PR

### CODE REVIEW

File: bin/lock.lua [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    unlockString = unlockString .. v .. '\n'
+end
+local unlockImage = paintutils.parseImage(unlockString)
+
+term.clear()
+
+local code = ...
+code = tonumber(code)
+
+print('Please place door or redstone to the right of this computer')
```
Comment (PoolloverNathan) [inline bin/lock.lua:43]: can this be configurable?

File: bin/lock.lua [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    if width == 7 and height == 5 then
+        monitor = v
+    end
+end
+
+if not monitor then
+    print('Please connect a 1x1 advanced monitor')
+    return;
+end
```
Comment (PoolloverNathan) [inline bin/lock.lua:66]: is it possible to support other resolutions with e.g. scaling? also, what if multiple monitors are connected?

File: bin/quarry.lua [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    keepItems = false
+else
+    print('Please specify if I should keep junk items (e.g. cobblestone) (true/false, arg 2)')
+    return
+end
+
+if not turtle then
+    print('This must be run from a turtle')
+    return
+end
```
Comment (PoolloverNathan) [inline bin/quarry.lua:21]: this assertion should probably be moved sooner

File: bin/quarry.lua [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    'copper',
+    'gold',
+    'redstone',
+    'emerald',
+    'lapis',
+    'diamond',
+    'debris',
+    'netherite',
+    'flint'
+}
```
Comment (PoolloverNathan) [inline bin/quarry.lua:35]: should add a mode that takes these as arguments (which I can do)

**Review by PoolloverNathan — CHANGES_REQUESTED** [review]

File: bin/lock.lua (hunk 2 above)
Comment (TheBunnyMan123) [reply bin/lock.lua:66]: because of how I coded it, not really. That is a good question, and I am ignoring it

File: bin/lock.lua (hunk 1 above)
Comment (TheBunnyMan123) [reply bin/lock.lua:43]: it can but I'm lazy

File: bin/quarry.lua (hunk 4 above)
Comment (TheBunnyMan123) [reply bin/quarry.lua:35]: sure, you can do that

File: bin/quarry.lua (hunk 3 above)
Comment (TheBunnyMan123) [reply bin/quarry.lua:21]: If you think it should be, feel free to, but I think it's fine

**Comment by PoolloverNathan** [comment]
going to put this on the to-code list for Saturday

File: .vscode/settings.json [hunk 5]
```diff
@@ -4,6 +4,7 @@
         "lowercase-global"
     ],
     "Lua.diagnostics.globals": [
+        "turtle"
```
Comment (PoolloverNathan) [inline .vscode/settings.json:7]: ±: missed comma
```suggestion
        "turtle",
```

File: bin/lock.lua [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+local lock = {
+    '  888  ',
+    ' 8   8 ',
+    ' 8   8 ',
+    '8888888',
+    '88fff88',
+    '888f888',
+    '888f888',
+    ' 88888 ',
+}
```
Comment (PoolloverNathan) [inline bin/lock.lua:12]: ±: needs to be updated to new system

File: bin/lock.lua [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    '  888  ',
+    ' 8   8 ',
+    ' 8   8 ',
+    ' 8     ',
+    '8888888',
+    '88fff88',
+    '888f888',
+    '888f888',
+    ' 88888 ',
+}
```
Comment (PoolloverNathan) [inline bin/lock.lua:31]: ±: see above

**Review by PoolloverNathan — CHANGES_REQUESTED** [review]
you changed code on main, have fun

**Review by PoolloverNathan — APPROVED** [review]
all ±s were resolved!

**Pull request merged**